custom_model_runner/datarobot_drum/resource/predict_mixin.py

```diff
@@ -53,10 +58,20 @@ def do_predict(self, logger=None):
         with tempfile.NamedTemporaryFile(suffix=file_ext) as f:
             filestorage.save(f)
             f.flush()
-            out_data = self._predictor.predict(f.name)
+            if self._target_type == TargetType.TRANSFORM:
+                out_data = self._predictor.transform(f.name)
+            else:
+                out_data = self._predictor.predict(f.name)
 
         if self._target_type == TargetType.UNSTRUCTURED:
             response = out_data
+        elif self._target_type == TargetType.TRANSFORM:
+            if is_sparse(out_data):
+                mtx_payload = make_mtx_payload(out_data)
+                response = '{{"X.transformed":{mtx_payload}}}'.format(mtx_payload=mtx_payload)
```

Comment:
You defined the key for X.transformed, but forgot to use it :)
We need to move it to the common.py, can be done later.

Reply:
:facepalm: